mitodl/micromasters
# Use database templates for faster database restore during selenium tests

#### What are the relevant tickets?
Fixes #3273 

#### What's this PR do?
 - Uses database templates for faster database restore. It previously dumped the database to SQL and reimported the SQL. This makes the `snapshot_dashboard_states` management command go roughly twice as fast.
 - Refactors the selenium tests to allow separate test cases
 - Adds a new function `setUpTestData` named after a similar function in `TestCase` which lets the user do some of the database setup per test case instead of per test
 - Fixes error handling for selenium test to leave the test database in a proper state on failure or exceptions
 - Adds `--suffix` to `snapshot_dashboard_states` to let the user generate screenshots for only a certain scenario
 - Adds `--list-scenarios` to `snapshot_dashboard_states` to list the scenarios which would be generated
 - I also made minor tweaks to the script which runs selenium tests to allow the use of `WEBPACK_DEV_SERVER_HOST`  in `.env`

#### How should this be manually tested?
Run the `snapshot_dashboard_states` management command and verify that it takes less than three minutes, roughly.
